core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2726,6 +2848,103 @@ public void testTableExistsFallbackToGETRequest() {
         ConfigResponse.builder().withEndpoints(ImmutableList.of(Endpoint.V1_LOAD_TABLE)).build());
   }
 
+  @Test
+  public void testSynchronousPlanningWithAllTasksInSingleResponse() throws IOException {
```

Comment:
nit: I'm aware that most tests in this class have a `test` prefix, but I feel like we could as well just omit that prefix from the newly added tests, to slightly improve readability, since that prefix doesn't add any value, wdyt?